### ISSUE TITLE
Clear test results on test execution?

### ISSUE THREAD

**Issue by ladariha** [jest-community/vscode-jest#1060]
### Environment

1. `vscode-jest version`: 6.0.1
2. `node -v`: 14.17.0
3. `npm -v` or `yarn --version`: yarn 1.22.19
4. `npm ls jest` or `npm ls react-scripts` (if you haven’t ejected): [fill]
5. your vscode-jest settings if customized:
    - jest.autoRun? off
    - jest.debugMode: true
    - jest.monitorLongRun: off
    - jest.autoClearTerminal: true
    - jest.virtualFolders : [{name: "somename", jestCommandLine: "/home/somePath/node_modules/jest/bin/jest.js --config=... "}]
    - anything else that you think might be relevant? [fill]
6. Operating system: RHEL 9

Would it be possible to clear previous test results (or make it configurable) when I start new test execution? What I mean is the right column in TEST RESULTS bottom panel. I'd find it much easier to use if the content of the history would be cleared anytime I trigger some tests. So essentially the desired state is to have history of size 1, aka latest test run only

![image](https://github.com/jest-community/vscode-jest/assets/611555/605030ff-5bce-4a42-9036-48b1b4e1edd0)

**Comment by abm0** [comment]
Is an option `"jest.autoClearOutput"` at v6.0.0 what you're looking for?

**Comment by connectdotz** [comment]
This is controlled by vscode testing subsystem, cc @connor4312 for insight.

**Comment by connor4312** [comment]
We could have a setting for this; if more users request it we could go that direction. An extension could also manually run the `testing.clearTestResults` command.

**Comment by connectdotz** [comment]
@connor4312 good point, we can try `testing.clearTestResults` upon starting new tests. 

One potential issue though: In a multi-root workspace environment, this will mean one run from a given folder will clear out the output from other folders... is there a way to close out the output associated with the specific controllers?

**Comment by connor4312** [comment]
That's right--unfortunately no command to clear a specific controller's output.

**Comment by ladariha** [comment]
Having it hidden behind some settings would be great :) 

Personally I don't really care about history of test runs - in my workflow, I don't really remember what exactly was different (either in application code or test code) in previous test executions so seeing different results brings zero value

**Comment by connor4312** [comment]
Yea, I think some `testing.historyEnabled` would be the right solution here. I would want to see if there's interest from more people here before taking the time to implement it (though I would take a PR)

**Comment by connor4312** [comment]
sounds good 🙂

**Comment by ladariha** [comment]
@connor4312 Here's the PR https://github.com/jest-community/vscode-jest/pull/1071 , hopefully somewhat meaningful - as stated in the PR's description, I did not bumped version in package.json and I kind of "guessed" the version in README.md file - not sure how to do it better? Thank you

**Comment by connor4312** [comment]
Oh, sorry. I was speaking to the VS Code side of things; I'm not a maintainer of the Jest extension. @connectdotz can weigh in on that PR. Code pointers for VS code if you opt to contribute there:

- Where configuration for testing is defined https://github.com/microsoft/vscode/blob/01c88cc99efcd423d662b56941f029bd22c4d584/src/vs/workbench/contrib/testing/common/configuration.ts
- Where results get inserted (could clear out incomplete results when a new result is inserted with the setting enabled) https://github.com/microsoft/vscode/blob/01c88cc99efcd423d662b56941f029bd22c4d584/src/vs/workbench/contrib/testing/common/testResultService.ts#L171

**Comment by ladariha** [comment]
No worries and sorry for the misunderstanding. I somehow managed to make the changes (unless proven otherwise in PR :))

**Comment by connectdotz** [comment]
@connor4312 after playing with `testing.clearTestResults` command, I am not sure that is what we are looking for. The command not only clears the test results and history in the panel, it also removes the status of all the test items. Did I miss something? I think we are looking for a way to clear just the output and the history of the runs, not the actual status of the tests.

**Comment by connectdotz** [comment]
reopen this issue to remind us when a better clear test-results panel command becomes available.

**Comment by connor4312** [comment]
The state of items is intrinsically linked to the history: the state of any given item is that of its most recent test run. There is not a way to keep states but remove the history, or vise versa.
